pkg/network/namescheme/networknamescheme.go

```diff
@@ -94,3 +97,34 @@ func generateOrdinalInterfaceName(idx int) string {
 	const ordinalIfacePrefix = "net"
 	return fmt.Sprintf("%s%d", ordinalIfacePrefix, idx)
 }
+
+// CreateNetworkNameSchemeByPodNetworkStatus create pod network name scheme according to the given VMI spec networks
+// and pod network status.
+// In case the pod network status has at least one interface with ordinal interface name it returns the ordinal network
+// name-scheme, Otherwise, returns the hashed network name scheme.
+func CreateNetworkNameSchemeByPodNetworkStatus(networks []v1.Network, podNetworkStatus map[string]networkv1.NetworkStatus) map[string]string {
+	if podHasOrdinalInterfaceName(podNetworkStatus) {
+		return CreateOrdinalNetworkNameScheme(networks)
+	}
+
+	return CreateNetworkNameScheme(networks)
+}
+
+// podHasOrdinalInterfaceName check if the given pod network status has at least one pod interface with ordinal name
+func podHasOrdinalInterfaceName(podNetworkStatus map[string]networkv1.NetworkStatus) bool {
```

Comment:
I would also rename this func to something like `doesPodHave...` .

Reply:
I prefer to keep the current naming, it reads batter IMO: (if pod-has-ordinal-interface-name vs. if does pod-has-ordinal-interface-name).

Reply:
It does not read well with an `if` in front.